central/role/default.go

```diff
@@ -78,15 +84,21 @@ func getAccessScopeIncludeAllID() string {
 	return EnsureValidAccessScopeID("unrestricted")
 }
 
-// IsDefaultRoleName checks if a given role name corresponds to a default role.
-func IsDefaultRoleName(name string) bool {
-	return DefaultRoleNames.Contains(name)
+// IsDefaultRoleName checks if a given role corresponds to a default role.
+func IsDefaultRoleName(role *storage.Role) bool {
+	return role.GetTraits().GetOrigin() == storage.Traits_DEFAULT || DefaultRoleNames.Contains(role.GetName())
```

Comment:
we should probably update `role/datastore/datastore_test` with tests around traits(or at least have a TODO linking to the ticket where it will be done)

Reply:
I've added tests under `default_test.go` which should be enough.